bee/src/main/java/com/apisense/bee/services/EventObserver.java

```diff
@@ -0,0 +1,175 @@
+package com.apisense.bee.services;
+
+import android.os.SystemClock;
+import android.util.Log;
+import android.view.accessibility.AccessibilityEvent;
+
+import com.apisense.bee.utils.accessibilitySting.AccessibilityEventWrapper;
+
+import java.util.Stack;
+
+import io.apisense.sting.phone.system.WindowChangeDetectingService;
+
+/**
+ * Created by Mohammad Naseri
+ */
+
+public class EventObserver extends WindowChangeDetectingService{
+    private static final String TAG = EventObserver.class.getName();
+
+    private static OnAccessibilityEvent callback;
+    static Stack<StackElement> events = new Stack<>();
+    static StackElement lastEvent = null;
+    static String tempInput = "";
+    static AccessibilityEventWrapper tempWrapper;
+
+    static String editTextClass = "android.widget.EditText";
+
+    static String widgetClass = "android.widget";
+
+    @Override
+    public void onServiceConnected() {
+        Log.d(TAG, "onServiceConnected");
+
+        long t_start = SystemClock.uptimeMillis();
+        Log.d(TAG, "EXECUTION START-TIME: " + t_start);
+    }
+
+    @Override
+    public void onAccessibilityEvent(AccessibilityEvent event) {
+        super.onAccessibilityEvent(event);
+
+        if (event != null) {
```

Comment:
Here I am checking the event in the input method. Not the events stack.

Reply:
Indeed, but since you are `calling super.onAccessibilityEvent(event);` you don't need the check anyway since the argument here is guaranteed to not be null.

It would remove a level of indent, thus being more easy to read.

Reply:
Good point!, removed it.